### PR TITLE
Add Maelstrom node target

### PR DESCRIPTION
**Pull request by makadaw**
Add echo server and single node KV storage. KV storage does not support
distribution across nodes

Start https://github.com/makadaw/swift-raft/issues/7

Hey, @ktoso if you have a few minutes, can look at the changes?

### CODE REVIEW

File: Sources/maelstrom-node/main.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import NIO
+import Logging
+import Lifecycle
+
+
+class App {
+
+    func run() throws {
+        var logger = Logger(label: "RAFT") { _ in
+            StreamLogHandler.standardError(label: "RAFT")
```
Comment (ktoso) [inline Sources/maelstrom-node/main.swift:15]: oh i see, it looks at stderr? cool :)

**Comment by ktoso** [comment]
Super cool, I'll give it a proper look :-)

File: Sources/MaelstromRaft/RPC/MaelstromRPC.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        do {
+            try group.syncShutdownGracefully()
+        } catch {
+            logger.error("Error shutting down: \(error)")
+            exit(0)
+        }
+        logger.info("Maelstrom stopped")
+    }
+
+}
```
Comment (ktoso) [inline Sources/MaelstromRaft/RPC/MaelstromRPC.swift:128]: very cool 👍

File: Sources/MaelstromRaft/RPC/RPCPacket.swift [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        return decoder
+    }
+
+    static var encoder: JSONEncoder {
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+        return encoder
+    }
```
Comment (ktoso) [inline Sources/MaelstromRaft/RPC/RPCPacket.swift:108]: 👍

**Review by ktoso — APPROVED** [review]
This is looking great! Very excited to use Maelstorm with this and other Swift projects

File: Sources/maelstrom-node/main.swift (hunk 1 above)
Comment (makadaw) [reply Sources/maelstrom-node/main.swift:15]: Yes, maelstrom read STDERR for logs

**Pull request merged**